shared/js/background/amp-protection.es6.js

```diff
@@ -0,0 +1,207 @@
+const Site = require('./classes/site.es6')
+const tdsStorage = require('./storage/tds.es6')
+
+let regexPatterns = null
+let keywords = null
+
+const featureName = 'ampLinks'
+
+/**
+ * Ensure the config has the ampLinks feature and its settings
+ *
+ * @returns {boolean} - true if the config is loaded
+ */
+function ensureConfig () {
+    if (regexPatterns && keywords) {
+        return true
+    }
+
+    if (!tdsStorage.config ||
+        !tdsStorage.config.features ||
+        !tdsStorage.config.features.ampLinks ||
+        !tdsStorage.config.features.ampLinks.settings ||
+        !tdsStorage.config.features.ampLinks.settings.linkFormats ||
+        !tdsStorage.config.features.ampLinks.settings.keywords) {
+        return false
+    }
+
+    regexPatterns = tdsStorage.config.features.ampLinks.settings.linkFormats
+    keywords = tdsStorage.config.features.ampLinks.settings.keywords
```

Comment:
Can you just make this a single var?
```suggestion
    ampSettings = tdsStorage.config.features.ampLinks.settings
```

That way checking and unsetting will be a little less error prone.